### PR TITLE
refactor: Remove Postgres env var

### PR DESCRIPTION
**Pull request by connorgorman**
## Description

Removes usages of the Postgres env var. Please review closely, lots of if cases

## Checklist
- [ ] Investigated and inspected CI test results
- [ ] Unit test and regression tests added
- [ ] Evaluated and added CHANGELOG entry if required
- [ ] Determined and documented upgrade steps
- [ ] Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))

If any of these don't apply, please comment below.

## Testing Performed

CI

### CODE REVIEW

File: central/graphql/resolvers/vulnerability_requests_test.go [hunk 1]
```diff
@@ -153,10 +152,8 @@ func (s *VulnRequestResolverTestSuite) TearDownTest() {
 }
 
 func (s *VulnRequestResolverTestSuite) SetupSuite() {
-	if env.PostgresDatastoreEnabled.BooleanSetting() {
-		s.T().Skip("Skip non postgres store tests")
-		s.T().SkipNow()
-	}
+	s.T().Skip("Skip non postgres store tests")
```
Comment (dashrews78) [inline central/graphql/resolvers/vulnerability_requests_test.go:155]: I assume this test is covered with one of the other tickets @md2119 has?

File: central/policy/datastore/datastore_impl_test.go [hunk 2]
```diff
@@ -44,9 +44,7 @@ type PolicyDatastoreTestSuite struct {
 }
 
 func (s *PolicyDatastoreTestSuite) SetupTest() {
-	if env.PostgresDatastoreEnabled.BooleanSetting() {
-		s.T().Skip("Tests are not applicable when ROX_POSTGRES_DATASTORE is set to true")
-	}
+	pgtest.SkipIfPostgresEnabled(s.T())
```
Comment (dashrews78) [inline central/policy/datastore/datastore_impl_test.go:47]: Add TODO if this test will be updated or removed separately.  I'm guessing ROX-17689

File: central/risk/multipliers/deployment/service_account_permissions.go [hunk 3]
```diff
@@ -53,55 +52,53 @@ func NewSAPermissionsMultiplier(roleStore roleStore.DataStore, bindingStore bind
 // Score takes a deployment and evaluates its risk based on the permissions granted to the deployment's service account
 func (c *saPermissionsMultiplier) Score(ctx context.Context, deployment *storage.Deployment, _ map[string][]*storage.Risk_Result) *storage.Risk_Result {
```
Comment (dashrews78) [inline central/risk/multipliers/deployment/service_account_permissions.go:53]: Remove?

File: central/risk/multipliers/deployment/service_account_permissions.go [hunk 4]
```diff
@@ -28,10 +27,10 @@ var (
...
 	maxPermissionScore = 1518.0
 
-	riskFactorPrefix     = "Service account %q has been granted"
-	clusterAdminSuffix   = "cluster admin privileges in the cluster"
-	clusterScopeSuffix   = "the following permissions on resources in the cluster:"
-	namespaceScopeSuffix = "the following permissions on resources in namespace %q:"
+	//riskFactorPrefix     = "Service account %q has been granted"
+	//clusterAdminSuffix   = "cluster admin privileges in the cluster"
+	//clusterScopeSuffix   = "the following permissions on resources in the cluster:"
+	//namespaceScopeSuffix = "the following permissions on resources in namespace %q:"
```
[the remark this answers is not in the document]
Comment (connorgorman) [reply central/risk/multipliers/deployment/service_account_permissions.go:33]: There's a ticket to fix this, so if we ever get to it then we would probably still need these

File: central/policy/datastore/datastore_impl_test.go (hunk 2 above)
Comment (connorgorman) [reply central/policy/datastore/datastore_impl_test.go:47]: yeah this will need to be addressed with policies

**Review by dashrews78 — APPROVED** [review]
Not a huge fan of the commented out code, but I don't have a better idea

**Review by md2119 — APPROVED** [review]
Please add a changelog for dropped RBAC risk.

**Pull request merged**